ci/validate.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+# This script is intended to be run by Cirrus-CI to validate PR
+# content prior to building any images.  It should not be run
+# under any other context.
+
+set -eo pipefail
+
+SCRIPT_FILEPATH=$(realpath "${BASH_SOURCE[0]}")
+SCRIPT_DIRPATH=$(dirname "$SCRIPT_FILEPATH")
+REPO_DIRPATH=$(realpath "$SCRIPT_DIRPATH/../")
+
+# shellcheck source=./lib.sh
+source "$REPO_DIRPATH/lib.sh"
+
+req_env_vars CIRRUS_PR CIRRUS_BASE_SHA
+
+# die() will add a reference to this file and line number.
+[[ "$CIRRUS_CI" == "true" ]] || \
+  die "This script is only/ever intended to be run by Cirrus-CI."
+
+for target in image_builder/gce.json base_images/cloud.json \
+              cache_images/cloud.json win_images/win-server-wsl.json; do
+  if ! make $target; then
+    die "Running 'make $target' failed, please validate input YAML files."
+  fi
+done
+
+# Variable is defined by Cirrus-CI at runtime
+# shellcheck disable=SC2154
+if ! git diff --name-only ${CIRRUS_BASE_SHA}..HEAD | grep -q IMG_SFX; then
+  die "Every PR must include an updated IMG_SFX file.
+Simply run 'make IMG_SFX', commit the result, and re-push."
+fi
+
+# Verify new IMG_SFX value always sorts later than previous value.  This prevents
+# screwups due to local timezone, bad, or unset clocks, etc.
+cd $REPO_DIRPATH
+# Automation requires the date and time components sort properly
+# as if they were version-numbers regardless of distro-version component.
+new_img_ver=$(awk -F '-' '{print $1"."$2}' ./IMG_SFX)
+# TODO: Conditional checkout not needed after the PR which added IMG_SFX file.
+if git checkout ${CIRRUS_BASE_SHA} IMG_SFX; then
```

Comment:
This `checkout` makes me uncomfortable: it leaves behind a bad file which may bite someone down the line. It's probably okay, ... unless someone adds a new YAML step after `ci/validate.sh`.

Would you consider using
```bash
v_prev=$(git show ${CIRRUS_BASE_SHA}:IMG_SFX 2>&1 || true) 
if ! [[ $v_prev =~ "exists on disk, but not in" ]]; then
    ...
```

Reply:
I could but...this is hurting my brain at the moment :grin:   I think my intention was (after this merges) to immediately remove the `if` so it always does the checkout.

Are you suggesting that at the end, after validating (good or bad), the script should `git checkout HEAD IMG_SFX` to make the repo. pristine?  Assuming so, I could live with that.  Opened https://github.com/containers/automation_images/pull/251

Reply:
No, my suggestion was never to do the unsafe checkout in the first place...

Reply:
But yeah, that can be done in a followup PR. If you want to merge this, I'm OK with that.

Reply:
Okay, made a note of this in followup PR.